packages/gsul/src/dynamoDBclient.ts

```diff
@@ -0,0 +1,61 @@
+
+import {DynamoDBDocumentClient, QueryCommand, QueryCommandInput} from "@aws-sdk/lib-dynamodb"
+import {Logger} from "@aws-lambda-powertools/logger"
+import{DynamoDBResult} from "./schema/result.ts"
+import {NativeAttributeValue} from "@aws-sdk/util-dynamodb"
+
+export function runDynamoDBQueries (queryParams: Array<QueryCommandInput>,
+  docClient: DynamoDBDocumentClient,
+  logger: Logger): Array<Promise<Array<DynamoDBResult>>> {
+
+  // helper function to deal with pagination of results from dynamodb
+  const getAllData = async (query: QueryCommandInput) => {
+    const _getAllData = async (query: QueryCommandInput, startKey: Record<string, NativeAttributeValue>) => {
+      if (startKey) {
+        query.ExclusiveStartKey = startKey
+      }
+      const command = new QueryCommand(query)
+      logger.info("running query", {query})
+      return docClient.send(command)
+    }
+    let lastEvaluatedKey = null
+    let rows = []
+    do {
+      const result = await _getAllData(query, lastEvaluatedKey)
+      rows = rows.concat(result.Items)
+      lastEvaluatedKey = result.LastEvaluatedKey
+    } while (lastEvaluatedKey)
```

Comment:
I don't think any of this is necessary it we do it like this https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/dynamodb-example-query-scan.html